Makefile

```diff
@@ -216,7 +216,7 @@ version:
 jar:
 	mill -i xiangshan.assembly
 
-$(JAR):
+$(JAR): FORCE
```

Comment:
If the dependency is empty, I thought `make` will always re-run it?

Reply:
the `$(JAR)` was a file target, which will not be re-built when target file exist :(

Reply:
I do not like the approach of marking a file target as PHONY. Making it depend on a PHONY target can achieve the same effect (refer to the FORCE in linux Kbuild).
When adding similar targets(e.g. the LIBSO target) later, there's no need to update the PHONY list - just have them depend on this FORCE target